Assingmenty/DiceRollGame/GameManager.cs

```diff
@@ -0,0 +1,34 @@
+﻿class GameManager
+{
+    private readonly int _randomNumber = RandomNumberProvider.Generate();
+    private readonly int _numberOfTries = 3;
+
+    public void RunGame()
+    {
+        bool isWon;
+        MessagePrinter.WelcomeUser(_numberOfTries);
+        isWon = PlayGame();
```

Comment:
domyślam się że w pythonie jest konwencja "wszystkie zmienne deklarujemy na górze metody" ale w C# nie ma. w tym przypadku możesz zrobić smiało deklarację w tej samej linijce co używasz PlayGame
czyli po prostu "bool isWon = PlayGame()"

Reply:
No właśnie w pythonie nie, ale w C# jest czasami ta potrzeba zadeklarowania przed użyciem i trochę nie czuję kiedy się powinno, a kiedy nie

Reply:
Tutaj pierwszy raz używasz tej zmiennej dopiero w linijce gdzie robisz PrintGameResult. to znaczy że przed tą linijką musi być zarówno zadeklarowana, jak i zainicjalizowana, żeby faktycznie miałą wartość. może to być tuż przed tą linijką, nie musi być z wyprzedzeniem żadnym. Ba, mogłoby to wręcz być:

MessagePrinter.PrintGameResult(PlayGame());

I wtedy by nie było w ogole zmiennej. Ale ze zmnienną jest ofc czytelniej.

Żeby nie było wątpliwości:
bool isWon; DEKLARACJA
isWon = false; INICJALIZACJA
bool isWon = false; JEDNOCZENA DEKLARACJA I INICJALIZACJA

w sumie zanotuję sobie żeby gdzieś na począku kursu pokazać analogiczną rozpiskę

Reply:
Wszystko jasne, dzięx.